wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java

```diff
@@ -358,6 +375,20 @@ public FileItemWrap getValue() {
 		return (FileItemWrap) getData();
 	}
 
+	/**
+	 * Retrieves an mime type of the uploaded file's contents.
+	 * This is not the content type passed by the browser.
+	 *
+	 * @return an file's mime type, or null if there was no file uploaded
+	 * @throws IOException if there is an error obtaining the input stream from the uploaded file.
+	 */
+	public String getMimeType() throws IOException {
+		if (getFile() != null) {
+			FileUtil.getFileMimeType(getFile());
+		}
+		return null;
+	}
+
```

Comment:
This function **always** returns `null`. Is that correct?

Reply:
No, if the widget has a file, then it's mime will be returned, otherwise null. 
I had added [additional class level javadoc](https://github.com/BorderTech/wcomponents/blob/4e6e825546737773489ce40ae9f377734e0e50c4/wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java#L38) indicating the new API, allowing the client to perform custom mime validations